tests/src/capability.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ This Source Code Form is subject to the terms of the Mozilla Public
+ License, v. 2.0. If a copy of the MPL was not distributed with this
+ file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
+
+
+/*!
+    \file test/capability.cpp
+    \brief Test optional capabilities
+ */
+
+#include "../common/testing.h"
+#include "../common/testUtils.h"
+
+#include <Ponca/src/Fitting/capabilities.h>
+#include <Ponca/src/Fitting/basket.h>
+#include <Ponca/src/Fitting/plane.h>
+#include <Ponca/src/Fitting/algebraicSphere.h>
+#include <Ponca/src/Fitting/weightFunc.h>
+#include <Ponca/src/Fitting/weightKernel.h>
+
+using namespace Ponca;
+
+constexpr int Dim = 3;
+using DataPoint = PointPosition<float, Dim>;
+using Kernel = ConstantWeightKernel<float>;
+using WeightFunc = DistWeightFunc<DataPoint, Kernel>;
+
+// compile-time and run-time check if a given value is true or false
+// this template function is required because "Outside a template, a discarded 
+// statement is fully checked." (cppreference.com/w/cpp/language/if) 
+template<bool b>
+void CHECK() {
+    static_assert(b);
+    VERIFY(b);
+}
+
+// extension that test if PLANE is provided
+template<class P, class W, typename T>
+class InitTestExtensionPlane : public T
+{
+public:
+    PONCA_FITTING_DECLARE_DEFAULT_TYPES
+    
+    inline void init(const VectorType& _basisCenter = VectorType::Zero())
+    {
+        Base::init(_basisCenter);
+        if constexpr( IS_PROVIDED(PLANE) ) {
+            CHECK<true>();
+            Base::setPlane(_basisCenter, VectorType::UnitZ());
+        } else {
+            // must not compile if PLANE is provided
+            CHECK<false>();
+        }
+
+        if constexpr( IS_PROVIDED(DOES_NOT_EXIST) ) {
+            // must not compile because DOES_NOT_EXIST does not exist
+            CHECK<false>();
+        }
```

Comment:
Can't you do something like this ?
```suggestion
        if constexpr( ) {
            // must not compile because DOES_NOT_EXIST does not exist
            CHECK<! IS_PROVIDED(DOES_NOT_EXIST)>();
        }
```

Reply:
Unfortunately no, because there is a lambda expression in `IS_PROVIDED` and using a lambda-expression in a template-argument is only available with c++20. 
Instead, I can do `static_assert(not IS_PROVIDED(DOES_NOT_EXIST));`, but this piece of code was more to show that the scope after the `if constexpr` is not compiled.